### PR TITLE
Fix ReblockGVCF NPE on no-calls

### PR DESCRIPTION
**Pull request by ldgauthier**

### CODE REVIEW

File: src/main/java/org/broadinstitute/hellbender/tools/walkers/annotator/InbreedingCoeff.java [hunk 1]
```diff
@@ -72,13 +73,14 @@ public InbreedingCoeff(final File pedigreeFile){
...
         if (genotypes == null || genotypes.size() < MIN_SAMPLES || !vc.isVariant()) {
+            logger.warn("InbreedingCoeff will not be calculated; at least " + MIN_SAMPLES + " samples must have called genotypes");
             return Collections.emptyMap();
         }
         final Pair<Integer, Double> sampleCountCoeff = calculateIC(vc, genotypes);
         final int sampleCount = sampleCountCoeff.getLeft();
         final double F = sampleCountCoeff.getRight();
         if (sampleCount < MIN_SAMPLES) {
-            logger.warn("Annotation will not be calculated, must provide at least " + MIN_SAMPLES + " samples");
+            logger.warn("InbreedingCoeff will not be calculated for at least one position; at least " + MIN_SAMPLES + " samples must have called genotypes");
```
Comment (droazen) [inline src/main/java/org/broadinstitute/hellbender/tools/walkers/annotator/InbreedingCoeff.java]: Did you want separate "one shot" loggers for these 2 warnings? (ie., would you ever want both warnings to be emitted?)

File: src/test/java/org/broadinstitute/hellbender/tools/walkers/variantutils/ReblockGVCFUnitTest.java [hunk 2]
```diff
@@ -71,6 +71,11 @@ public void testLowQualVariantToGQ0HomRef() {
         Assert.assertTrue(modified.getAlternateAllele(0).equals(Allele.NON_REF_ALLELE));
         Assert.assertTrue(!modified.filtersWereApplied());
         Assert.assertTrue(modified.getLog10PError() == VariantContext.NO_LOG10_PERROR);
+
+        final Genotype g2 = makeG("sample1", Allele.NO_CALL,Allele.NO_CALL);
+        final VariantContext noData = makeDeletionVC("noData", Arrays.asList(LONG_REF, DELETION, Allele.NON_REF_ALLELE), LONG_REF.length(), g2);
+        final VariantContext notCrashing = reblocker.lowQualVariantToGQ0HomRef(noData, noData);
+        Assert.assertTrue(notCrashing.getGenotype(0).isNoCall());
```
Comment (droazen) [inline src/test/java/org/broadinstitute/hellbender/tools/walkers/variantutils/ReblockGVCFUnitTest.java]: Add a comment explaining this test case (ie., explain what the "crash" being tested for was)

File: src/test/java/org/broadinstitute/hellbender/tools/walkers/variantutils/ReblockGVCFUnitTest.java [hunk 3]
```diff
@@ -71,6 +71,11 @@ public void testLowQualVariantToGQ0HomRef() {
...
     }
 
     @Test
```
Comment (droazen) [inline src/test/java/org/broadinstitute/hellbender/tools/walkers/variantutils/ReblockGVCFUnitTest.java]: Can you add a test case covering the change to `shouldBeReblocked()` as well (and make that method `@VisibleForTesting`?)

File: src/main/java/org/broadinstitute/hellbender/tools/walkers/annotator/InbreedingCoeff.java (hunk 1 above)
Comment (ldgauthier) [reply src/main/java/org/broadinstitute/hellbender/tools/walkers/annotator/InbreedingCoeff.java]: The first one is like "hey, you don't have enough samples in your cohort" and the second one is like "oops, there are a lot of no-calls at this site, so I'm not going to output a value here", so they should be mutually exclusive.

**Comment by ldgauthier** [comment]
@droazen I addressed your comments, but I also added some more fixes (and tests!)

**Comment by droazen** [comment]
Closing this one -- these changes to `ReblockGVCFs` went in as part of https://github.com/broadinstitute/gatk/pull/4947

**Comment by ldgauthier** [comment]
This got merged into #4947

**Pull request closed without merging**